cypress/integration/task_metadata_card.js

```diff
@@ -0,0 +1,52 @@
+/// <reference types="Cypress" />
+import { waitForGQL } from "../utils/networking";
+import get from "lodash/get";
+const taskId =
+  "evergreen_ubuntu1604_test_model_patch_5e823e1f28baeaa22ae00823d83e03082cd148ab_5e4ff3abe3c3317e352062e4_20_02_21_15_13_48";
+const taskRoute = `/task/${taskId}`;
+const taskRouteWithoutDependsOn = `/task/evergreen_ubuntu1604_test_migrations_patch_5e823e1f28baeaa22ae00823d83e03082cd148ab_5e4ff3abe3c3317e352062e4_20_02_21_15_13_48`;
+
+describe("Task Metadata Card", function() {
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+    cy.route("POST", "/graphql/query").as("gqlQuery");
+  });
+
+  it("Should show an error message when navigating to a nonexistent task id", () => {
+    cy.visit("task/not-real");
+    cy.get("[data-cy=metadata-card-error]").should("exist");
+  });
+
+  it("Base commit link should have href", () => {
+    cy.visit(taskRoute);
+    cy.get("[data-cy=base-task-link]").should("have.attr", "href");
+  });
+
+  it("Host link should have href", () => {
+    cy.visit(taskRoute);
+    cy.get("[data-cy=task-host-link]").should("have.attr", "href");
+  });
+
+  it("Depends on task should have href", () => {
+    cy.visit(taskRoute);
+    cy.get("[data-cy=depends-on-link]").should("have.attr", "href");
+  });
```

Comment:
test would still pass if the href pointed to hotmail.com. `.should("have.attr", "href")` and `.include("/taskity/task")`

Reply:
the href should look like this `task/{taskId}`. It's impossible to infer what the taskId would be based on any information from the ui route so I can't match on that. 
I thought of matching on `task` but that information comes from GQL and it seems that we are testing the data instead of functionality. 
It crossed my mind that generating links to spruce pages should be contained inside Spruce and not GQL because the source of truth for our routes exists within Spruce.

Reply:
testing that it includes `/task` in the href is not testing the data because that value is constant. it is not generated by the server, it is just included in the field value out of convenience.